cadc-data-ops-fits/src/main/java/org/opencadc/fits/slice/ShapeCutout.java

```diff
@@ -94,6 +98,30 @@ public ShapeCutout(FITSHeaderWCSKeywords fitsHeaderWCSKeywords) {
         super(fitsHeaderWCSKeywords);
     }
 
+    /**
+     * Ensure the proper axis values are used in the case of a compressed HDU (ZNAXISn) or a regular Image HDU.
+     *
+     * @return the dimensions of the axis.  Returns an empty array for no NAXIS values.
+     */
+    int[] getDimensions() {
+        final int nAxis = this.fitsHeaderWCSKeywords.getIntValue(Compression.ZNAXIS.key(),
+                                                                 this.fitsHeaderWCSKeywords.getIntValue(
+                                                                         Standard.NAXIS.key()));
+
+        if (nAxis == 0) {
+            return new int[0];
+        }
+
+        final int[] axes = new int[nAxis];
+        for (int i = 1; i <= nAxis; i++) {
```

Comment:
Blindly using ZNAXISi or NAXISi here seems dangerous. If you actually used a value from ZNAXIS above, you should only use ZNAXISi here (and similarily for NAXIS and NAXISi). It could be that the problem only shows up in incorrect fits headers, but it would go unnoticed and probably make it harder to figure out if this code isn't more careful.

Reply:
OK.  I've clarified when it gets used.  Thanks.